### ISSUE TITLE
AI Ollama integration

### ISSUE THREAD

**Issue by mathman-git** [dreeveapp/dreeve#2513]
### Your question

I'm trying to get local Ollama working, but although I can succesfully curl from the dreeve docker to the ollama api, but when I configure it in Dreeve and try the chat, it gives a network booboo:

It looks like the url is not accepted?

Oh no, I made a booboo...
Network error during POST chat: URI must include a scheme and host. Use an absolute URI, a network-path reference starting with //, or configure a base_uri.: #0 /var/www/vendor/neuron-core/neuron-ai/src/HttpClient/GuzzleHttpClient.php(77): NeuronAI\HttpClient\GuzzleHttpClient->handleException(Object(NeuronAI\HttpClient\HttpRequest), Object(GuzzleHttp\Exception\RequestException)) #1 /var/www/vendor/neuron-core/neuron-ai/src/Providers/Ollama/HandleStream.php(49): NeuronAI\HttpClient\GuzzleHttpClient->stream(Object(NeuronAI\HttpClient\HttpRequest)) #2 /var/www/vendor/neuron-core/neuron-ai/src/Agent/Nodes/StreamingNode.php(50): NeuronAI\Providers\Ollama\Ollama->stream(Object(NeuronAI\Chat\Messages\UserMessage)) #3 /var/www/vendor/neuron-core/neuron-ai/src/Workflow/Executor/WorkflowExecutor.php(218): NeuronAI\Agent\Nodes\StreamingNode->__invoke(Object(NeuronAI\Agent\Events\AIInferenceEvent), Object(NeuronAI\Agent\AgentState)) #4 /var/www/vendor/neuron-core/neuron-ai/src/Workflow/Executor/WorkflowExecutor.php(167): NeuronAI\Workflow\Executor\WorkflowExecutor->executeNode(Object(App\Domain\Integration\AI\NeuronAIAgent), Object(NeuronAI\Agent\Events\AIInferenceEvent), Object(NeuronAI\Agent\Nodes\StreamingNode), NULL) #5 /var/www/vendor/neuron-core/neuron-ai/src/Workflow/Executor/WorkflowExecutor.php(77): NeuronAI\Workflow\Executor\WorkflowExecutor->traverseNodes(Object(App\Domain\Integration\AI\NeuronAIAgent), Object(NeuronAI\Agent\Events\AIInferenceEvent), Object(NeuronAI\Agent\Nodes\StreamingNode)) #6 /var/www/vendor/neuron-core/neuron-ai/src/Workflow/Workflow.php(183): NeuronAI\Workflow\Executor\WorkflowExecutor->run(Object(App\Domain\Integration\AI\NeuronAIAgent)) #7 [internal function]: NeuronAI\Workflow\Workflow->run() #8 /var/www/vendor/neuron-core/neuron-ai/src/Workflow/WorkflowHandler.php(46): Generator->valid() #9 /var/www/src/Controller/AIChatRequestHandler.php(121): NeuronAI\Workflow\WorkflowHandler->events() #10 /var/www/vendor/symfony/http-foundation/EventStreamResponse.php(66): App\Controller\AIChatRequestHandler->{closure:App\Controller\AIChatRequestHandler::chatSse():93}(Object(Symfony\Component\HttpFoundation\EventStreamResponse)) #11 /var/www/vendor/symfony/http-kernel/HttpKernel.php(104): Symfony\Component\HttpFoundation\EventStreamResponse->{closure:Symfony\Component\HttpFoundation\EventStreamResponse::setCallback():65}() #12 /var/www/vendor/symfony/http-foundation/StreamedResponse.php(125): Symfony\Component\HttpKernel\HttpKernel::{closure:Symfony\Component\HttpKernel\HttpKernel::handle():101}() #13 /var/www/vendor/symfony/http-foundation/Response.php(408): Symfony\Component\HttpFoundation\StreamedResponse->sendContent() #14 /var/www/vendor/symfony/runtime/Runner/FrankenPhpWorkerRunner.php(67): Symfony\Component\HttpFoundation\Response->send() #15 [internal function]: Symfony\Component\Runtime\Runner\FrankenPhpWorkerRunner->{closure:Symfony\Component\Runtime\Runner\FrankenPhpWorkerRunner::run():51}() #16 /var/www/vendor/symfony/runtime/Runner/FrankenPhpWorkerRunner.php(72): frankenphp_handle_request(Object(Closure)) #17 /var/www/vendor/autoload_runtime.php(32): Symfony\Component\Runtime\Runner\FrankenPhpWorkerRunner->run() #18 /var/www/public/index.php(5): require_once('/var/www/vendor...') #19 {main}

When I run the following commands locally within the container of dreeve, I get succesfully a reply:
curl http://dreeve-ollama:11434/api/generate -d '{ "model": "llama3.2", "prompt": "Hi" }'
curl http://<external-ip of box>:11434/api/generate -d '{ "model": "llama3.2", "prompt": "Hi" }'

URL which I enter in the GUI:
http://dreeve-ollama:11434/api
http://<external-ip of box>:11434/api

Thanks in advance!

**Comment by robiningelbrecht** [comment]
This is a bug, you should be able to configure the URL to Ollame in Dreeve; but that's not possible at the moment. Will be fixed in the next version

**Comment by mathman-git** [comment]
It isn't resolved yet. I see communication between the app container and ollama container (they reside in the same network), but after exactly a minute, there comes up another error: 

Network error during POST chat: Connection refused for URI http://dreeve-ollama:11434/api/chat: #0 /var/www/vendor/neuron-core/neuron-ai/src/HttpClient/GuzzleHttpClient.php(77): NeuronAI\HttpClient\GuzzleHttpClient->handleException(Object(NeuronAI\HttpClient\HttpRequest), Object(GuzzleHttp\Exception\ConnectException)) #1 /var/www/vendor/neuron-core/neuron-ai/src/Providers/Ollama/HandleStream.php(49): NeuronAI\HttpClient\GuzzleHttpClient->stream(Object(NeuronAI\HttpClient\HttpRequest)) #2 /var/www/vendor/neuron-core/neuron-ai/src/Agent/Nodes/StreamingNode.php(50): NeuronAI\Providers\Ollama\Ollama->stream(Object(NeuronAI\Chat\Messages\UserMessage)) #3 /var/www/vendor/neuron-core/neuron-ai/src/Workflow/Executor/WorkflowExecutor.php(218): NeuronAI\Agent\Nodes\StreamingNode->__invoke(Object(NeuronAI\Agent\Events\AIInferenceEvent), Object(NeuronAI\Agent\AgentState)) #4 /var/www/vendor/neuron-core/neuron-ai/src/Workflow/Executor/WorkflowExecutor.php(167): NeuronAI\Workflow\Executor\WorkflowExecutor->executeNode(Object(App\Domain\Integration\AI\NeuronAIAgent), Object(NeuronAI\Agent\Events\AIInferenceEvent), Object(NeuronAI\Agent\Nodes\StreamingNode), NULL) #5 /var/www/vendor/neuron-core/neuron-ai/src/Workflow/Executor/WorkflowExecutor.php(77): NeuronAI\Workflow\Executor\WorkflowExecutor->traverseNodes(Object(App\Domain\Integration\AI\NeuronAIAgent), Object(NeuronAI\Agent\Events\AIInferenceEvent), Object(NeuronAI\Agent\Nodes\StreamingNode)) #6 /var/www/vendor/neuron-core/neuron-ai/src/Workflow/Workflow.php(183): NeuronAI\Workflow\Executor\WorkflowExecutor->run(Object(App\Domain\Integration\AI\NeuronAIAgent)) #7 [internal function]: NeuronAI\Workflow\Workflow->run() #8 /var/www/vendor/neuron-core/neuron-ai/src/Workflow/WorkflowHandler.php(46): Generator->valid() #9 /var/www/src/Controller/AIChatRequestHandler.php(121): NeuronAI\Workflow\WorkflowHandler->events() #10 /var/www/vendor/symfony/http-foundation/EventStreamResponse.php(66): App\Controller\AIChatRequestHandler->{closure:App\Controller\AIChatRequestHandler::chatSse():93}(Object(Symfony\Component\HttpFoundation\EventStreamResponse)) #11 /var/www/vendor/symfony/http-kernel/HttpKernel.php(104): Symfony\Component\HttpFoundation\EventStreamResponse->{closure:Symfony\Component\HttpFoundation\EventStreamResponse::setCallback():65}() #12 /var/www/vendor/symfony/http-foundation/StreamedResponse.php(125): Symfony\Component\HttpKernel\HttpKernel::{closure:Symfony\Component\HttpKernel\HttpKernel::handle():101}() #13 /var/www/vendor/symfony/http-foundation/Response.php(408): Symfony\Component\HttpFoundation\StreamedResponse->sendContent() #14 /var/www/vendor/symfony/runtime/Runner/FrankenPhpWorkerRunner.php(67): Symfony\Component\HttpFoundation\Response->send() #15 [internal function]: Symfony\Component\Runtime\Runner\FrankenPhpWorkerRunner->{closure:Symfony\Component\Runtime\Runner\FrankenPhpWorkerRunner::run():51}() #16 /var/www/vendor/symfony/runtime/Runner/FrankenPhpWorkerRunner.php(72): frankenphp_handle_request(Object(Closure)) #17 /var/www/vendor/autoload_runtime.php(32): Symfony\Component\Runtime\Runner\FrankenPhpWorkerRunner->run() #18 /var/www/public/index.php(5): require_once('/var/www/vendor...') #19 {main}

And the logging in the ollama container:
srv  server_strea: conv_id= (empty=1)
slot get_availabl: id  0 | task -1 |  - checking sim = 0.221 (512/2319) > 0.100
slot get_availabl: id  0 | task -1 | selected slot by LCP similarity, f_sim_best = 0.221 (> 0.100 thold), f_keep = 1.000
slot launch_slot_: id  0 | task -1 | sampler chain: logits -> ?penalties -> ?dry -> ?top-n-sigma -> top-k -> ?typical -> top-p -> ?min-p -> ?xtc -> temp-ext -> dist 
slot launch_slot_: id  0 | task -1 | sampler params: 
        repeat_last_n = 64, repeat_penalty = 1.000, frequency_penalty = 0.000, presence_penalty = 0.000
        dry_multiplier = 0.000, dry_base = 1.750, dry_allowed_length = 2, dry_penalty_last_n = 64
        top_k = 40, top_p = 0.900, min_p = 0.000, xtc_probability = 0.000, xtc_threshold = 0.100, typical_p = 1.000, top_n_sigma = -1.000, temp = 0.800
        mirostat = 0, mirostat_lr = 0.100, mirostat_ent = 5.000, adaptive_target = -1.000, adaptive_decay = 0.900
slot launch_slot_: id  0 | task 3 | processing task, is_child = 0
slot   operator(): id  0 | task 3 | new prompt, n_ctx_slot = 4096, n_keep = 4, task.n_tokens = 2319
slot   operator(): id  0 | task 3 | cached n_tokens = 512, memory_seq_rm [512, end)
[GIN] 2026/08/17 - 05:31:59 | 500 |          1m0s |      172.20.0.3 | POST     "/api/chat"
srv          stop: cancel task, id_task = 3
slot      release: id  0 | task 3 | stop processing: n_tokens = 1024, truncated = 0
srv  update_slots: all slots are idle

Maybe a timeout in dreeve of 2 minutes? I run it on a very slow box right now, so it takes its time to respond

**Comment by robiningelbrecht** [comment]
There's a default timeout of 60 seconds. If your box cannot answer in 60s, I'd increase the resources. The AI feature will never work properly with such a slow box
